src/db-api.js

```diff
@@ -66,10 +66,51 @@ const transactionsHistoryForAddresses = (db: Pool) => async (
   dateFrom: Date,
 ): Promise<ResultSet> => db.query(txHistoryQuery(limit), [addresses, dateFrom]);
 
+// The remaining queries should be used only for the purposes of the legacy API!
+
+/**
+ * Queries DB looking for successful transactions associated with the given address
+ * @param {Db Object} db
+ * @param {Address} address
+ */
+const addressSummary = (db: Pool) => async (address: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT * FROM "txs" WHERE hash = ANY (SELECT tx_hash from "tx_addresses" WHERE address = $1) AND tx_state = $2',
+    values: [address, 'Successful'],
+  });
+
+/**
+* Queries TXS table looking for a successful transaction with a given hash
+* @param {Db Object} db
+* @param {*} tx
+*/
+const txSummary = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT * FROM "txs" WHERE hash = $1 AND tx_state = $2',
```

Comment:
I would hardcode the "successful" string into the query, no need to pass it as a parameter